eth/borfinality/whitelist_helpers.go

```diff
@@ -86,13 +91,17 @@ func fetchWhitelistMilestone(ctx context.Context, heimdall heimdall.IHeimdallCli
 	return num, hash, nil
 }
 
-func fetchNoAckMilestone(ctx context.Context, heimdall heimdall.IHeimdallClient) (string, error) {
+func fetchNoAckMilestone(ctx context.Context, heimdallClient heimdall.IHeimdallClient) (string, error) {
```

Comment:
you can add logger log.Logger as a param

this assumes the call would become: fetchNoAckMilestoneByID(ctx, heimdall, milestoneID, config.logger)